src/component/handlers/edit/DraftEditorEditAndroidHandler.js

```diff
@@ -12,14 +12,35 @@
 
 'use strict';
 
+const EditorState = require('EditorState');
+
 const onBeforeInput = require('editOnBeforeInputAndroid');
 const onCompositionStart = require('editOnCompositionStart');
 const onSelect = require('editOnSelect');
+const onCopy = require('editOnCopy');
+const onPaste = require('editOnPaste');
+
+// For Android, we need to re-force the selection immediately outside the event to get the caret in 
+// the proper position and to re-trigger the keyboard.
+function onPasteWithSelection(editor, e) {
+  onPaste(editor, e);
+
+  // Force a reset of the selection on the next tick.
+  // `forceSelection` is needed or else the selection will be at the start of the pasted fragment.
+  // `setImmediate` is needed or else the keyboard will close.
+  setImmediate(() => {
+    editor.update(
+      EditorState.forceSelection(editor._latestEditorState, editor._latestEditorState.getSelection()),
+    );
+  });
+}
 
-// Noop everything except beforeInput and select...
+// Only handle the following events:
 const DraftEditorEditAndroidHandler = {
   onBeforeInput,
   onCompositionStart,
+  onCopy,
+  onPaste: onPasteWithSelection,
```

Comment:
Just to check, if they click into the middle of a word and paste stuff, the composition ends before the paste stuff goes through? Or do we need to add this to the android composition handler too?

Reply:
I believe so… HOWEVER it's really hard to paste without a selection.
Pasting into an empty block seems to work though.